### PR TITLE
Staging Deployment

### PR DESCRIPTION
**Pull request by evmiguel**
This PR is setting up the backend to deal with Heroku integration, particularly with the Heroku PostgreSQL database

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -29,6 +32,7 @@ app.use(
 
 app.use(authRouter);
 app.use('/media', s3Router);
+app.use(express.static('../../dist'));
```
Comment (evmiguel) [inline src/server/index.js]: @gnarf @ktbee Is the app going to be server side rendered? If so, routing doesn't work right now.

File: README.md [hunk 2]
```diff
@@ -109,3 +103,26 @@ aws_secret_access_key = <your secret access key id>
...
 ```
+
+## Deployment
+This app is deployed via Heroku. To run the deploy commands, please [install Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli).
+
+
+### Staging environment setup
+
+#### Initializing staging database
+*This is a temporary workaround* First, replace the values for user, password, host, and database in `src/server/database.json`. Once those values have been added, run:
```
Comment (evmiguel) [inline README.md]: Wanted to highlight this temporary workaround for setting up the staging database.

File: src/server/index.js (hunk 1 above)
Comment (gnarf) [reply src/server/index.js]: It needs that "default 404 page is index.html" thing to work

Re: SSR - doubtful - we might do it as a feature longer term, but I would almost see this "server" as having nothing to do with the frontend... I.E. not even doing this express.static longer term...

The long term goal IMO:  This is an "API" server, there is a "frontend" and the "api" server and they are just mounted at different directories on the "front server"...

I.E. lets let nginx / etc do the html / dist serving and then proxy requests to /api/whatever to the api server.

As a "yarn start" frontend, you can use a proxy on the frontend's webpack to proxy to the api server for /api routes.


All this "longer term" stuff doesn't need to happen >now< but you should have its concepts ready to go sooner or later.

File: README.md (hunk 2 above)
Comment (ktbee) [reply README.md]: Thanks for highlighting! I'm glad you pointed that out, I'll add a task to set up some env variables handling. I think having a programmatic way for determining what those are will solve this and a couple other issues (needing to export variables on the command line, having a secret committed to our repo).

File: README.md [hunk 3]
```diff
@@ -17,12 +17,6 @@ Follow the instructions here: [https://github.com/bocoup/threeflows#local-databa
 
 This is the list of all environment variables that should be exported prior to running the Teacher Moments server in the command line:
 ```
-export PGUSER=
-export PGPASSWORD=
-export PGDATABASE=
-export PGHOST=
-export PGPORT=
```
Comment (ktbee) [inline README.md]: Do we still need to export these elsewhere? Or are they covered in database.json?

File: README.md (hunk 2 above)
Comment (ktbee) [reply README.md]: Oh wait, looks like we're using `dotenv`? That might get us the tools we need to solve this issue in the long term.

**Comment by ktbee** [comment]
Oh and I also forgot to mention: looks like there's some merge and linting errors for this PR. I found `yarn prettier` fixed the linting issues though.

**Comment by rwaldron** [comment]
@ktbee @gnarf despite several attempts to rebase the commits in this pull request, I was unsuccessful. I then attempted to piece together the relevant bits to create a staging deployment, but was unsuccessful.

**Comment by ktbee** [comment]
We ended up covering a lot of these changes in #57. Thanks for getting us started on this @evmiguel!

**Pull request closed without merging**